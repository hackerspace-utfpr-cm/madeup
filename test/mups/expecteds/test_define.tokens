1(0-1) TO: to
1(3-7) ID: plus1
1(9-9) ID: x
1(10-10) NEWLINE: \n
2(13-13) ID: x
2(15-15) PLUS: +
2(17-17) INTEGER: 1
2(18-18) NEWLINE: \n
3(19-21) END: end
3(22-22) NEWLINE: \n
4(23-23) NEWLINE: \n
5(24-28) ID: print
5(30-34) ID: plus1
5(36-37) INTEGER: 10
5(38-38) NEWLINE: \n
6(39-43) ID: print
6(45-49) ID: plus1
6(51-52) INTEGER: -5
6(53-53) NEWLINE: \n
7(54-58) ID: print
7(60-64) ID: plus1
7(66-66) INTEGER: 0
7(67-67) NEWLINE: \n
8(68-72) ID: print
8(74-78) ID: plus1
8(80-82) INTEGER: 100
8(83-83) NEWLINE: \n
9(84-84) END_OF_FILE: END OF FILE
