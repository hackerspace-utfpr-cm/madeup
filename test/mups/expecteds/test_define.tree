(block (define plus1 ((x)) (block (+ (call-with-names x) (INTEGER 1)))) (call print (call plus1 (INTEGER 10))) (call print (call plus1 (INTEGER -5))) (call print (call plus1 (INTEGER 0))) (call print (call plus1 (INTEGER 100))))
