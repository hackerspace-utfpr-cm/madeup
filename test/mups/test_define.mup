to plus1 x
  x + 1
end

print plus1 10
print plus1 -5
print plus1 0
print plus1 100
