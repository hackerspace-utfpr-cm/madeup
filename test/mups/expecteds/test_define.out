11
-4
1
101
